CHANGELOG.md

```diff
@@ -4,6 +4,376 @@ All notable changes to Router will be documented in this file.
+
+ - build docker images from arbitrary (nightly) builds (-a)
+ - build an amd64 docker image on an arm64 machine (or vice versa) (-m)
+ - change the name of the image from the default 'router' (-n)
+
+Note: the build machine image architecture is used if the -m flag is not supplied.
+
+By [@garypen](https://github.com/garypen) in https://github.com/apollographql/router/pull/3304
+
+## 🐛 Fixes
+
+### Bring root span name in line with otel semantic conventions. ([Issue #3229](https://github.com/apollographql/router/issues/3229))
+
+Root span name has changed from `request` to `<graphql.operation.kind> <graphql.operation.name>`
+
+[Open Telemetry graphql semantic conventions](https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/instrumentation/graphql/) specify that the root span name must match the operation kind and name. 
+
+Many tracing providers don't have good support for filtering traces via attribute, so this change will bring significant usability enhancements to the tracing experience.
+
+By [@BrynCooke](https://github.com/BrynCooke) in https://github.com/apollographql/router/pull/3364
+
+### An APQ query with a mismatched hash will error as HTTP 400 ([Issue #2948](https://github.com/apollographql/router/issues/2948))
+
+We now model the both the behavior of the Gateway and the intended behavior of [the implementation](https://www.apollographql.com/docs/apollo-server/performance/apq/).  Even if our previous behavior was still acceptable, any other behavior is a misconfiguration of a client and should be prevented early.
+
+Previously, if a client sent an operation with an APQ hash, we would merely log an error to the console, **not** register the operation (for the next request) but still execute the query.  We now return a GraphQL error and don't execute the query.  No clients should be impacted by this, though anyone who had hand-crafted a query **with** APQ information (for example, copied a previous APQ-registration query but only changed the operation without re-calculating the SHA-256) might now be forced to use the correct hash (or more practically, remove the hash).
+
+By [@o0Ignition0o](https://github.com/o0Ignition0o) in https://github.com/apollographql/router/pull/3128
+
+### fix(subscription): take the callback url path from the configuration ([Issue #3361](https://github.com/apollographql/router/issues/3361))
+
+Previously when you specified the `subscription.mode.callback.path` it was not used, we had an hardcoded value set to `/callback`. It's now using the specified path in the configuration
+
+By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/3366
+
+### Preserve all shutdown receivers across reloads ([Issue #3139](https://github.com/apollographql/router/issues/3139))
+
+Preserve a vec of all channels that we create and process all of them during shutdown. This will avoid scenarios such as:
```

Comment:
```suggestion
We keep a list of all currently in use router configurations and active requests and process all of them during shutdown. This will avoid scenarios such as:
```